diesel/src/query_dsl/positional_order_dsl.rs

```diff
@@ -111,22 +111,12 @@ impl Order for Desc<OrderColumn> {
     }
 }
 
-macro_rules! impl_order_for_all_tuples {
-    ($(
-        $unused1:tt {
-            $(($idx:tt) -> $T:ident, $unused2:ident, $unused3:tt,)+
-        }
-    )+) => {
-        $(
-            impl<$($T: Order),+> Order for ($($T,)+) {
-                type Fragment = ($(<$T as Order>::Fragment,)+);
-
-                fn into_fragment(self) -> Self::Fragment {
-                    ($(self.$idx.into_fragment(),)+)
-                }
-            }
-        )+
-    };
-}
+#[diesel_derives::__diesel_for_each_tuple]
+impl<#[repeat] T: Order> Order for (T,) {
+    type Fragment = (<T as Order>::Fragment,);
 
-__diesel_for_each_tuple!(impl_order_for_all_tuples);
+    fn into_fragment(self) -> Self::Fragment {
+        #[repeat]
```

Comment:
I felt this was inconsistent and was thinking this `#[repeat]` should be inside the `()` on the item because we were using this to repeat whole statements. But then I saw `FromStaticSqlRow` where 2 items are supposed to repeat. I don't mind the exception now, but I would still suggest looking into making this a bit better for consistency.